samples/client/petstore/swift6/urlsessionLibrary/SwaggerClientTests/SwaggerClient/BearerTokenHandler.swift

```diff
@@ -12,21 +12,28 @@ import PetstoreClient
 public class BearerOpenAPIInterceptor: OpenAPIInterceptor {
     public init() {}
     
-    public func intercept(urlRequest: URLRequest, urlSession: URLSessionProtocol, openAPIClient: OpenAPIClient, completion: @escaping (Result<URLRequest, any Error>) -> Void) {
-        refreshTokenIfDoesntExist { token in
+    public func intercept<T>(urlRequest: URLRequest, urlSession: URLSessionProtocol, requestBuilder: RequestBuilder<T>, completion: @escaping (Result<URLRequest, any Error>) -> Void) {
+
+        guard requestBuilder.requiresAuthentication else {
+            // no authentication required
+            completion(.success(urlRequest))
+            return
+        }
+
+        BearerTokenHandler.shared.refreshTokenIfDoesntExist { token in
```

Comment:
@Jonas1893 here it's just `refreshTokenIfDoesntExist { token in`

Reply:
That's what I usually do 👍

Reply:
That's weird, when I ran the script locally, it showed the error.
The `swift6_test_all.sh` starts by running some unit tests with a sample Xcode project, not SPM.

```
# example project with unit tests
(cd $DIRECTORY/alamofireLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/apiNonStaticMethod/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/asyncAwaitLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
# (cd $DIRECTORY/combineLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/combineDeferredLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/default/SwaggerClientTests/ && ./run_xcodebuild.sh)
# (cd $DIRECTORY/promisekitLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/rxswiftLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
(cd $DIRECTORY/urlsessionLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)
```

By the way, CI is now green, so I will merge this PR, but we can continue to discuss some things here if you still have problems 👍

Reply:
You are right, when running `(cd $DIRECTORY/urlsessionLibrary/SwaggerClientTests/ && ./run_xcodebuild.sh)` isolated I see the error. I think the issue on my machine is that the script `swift6_test_all.sh` continues to execute all builds and the one test build failure got lost in the bash history since the script does not exit immediately if there is a build failure in one test run.

Thx for accepting this!

Reply:
If you have any suggestion to improve this, feel free to create a PR or issue on this 👍 
Your contributions are welcomed 👍